metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py

```diff
@@ -879,6 +929,153 @@ def _extract_lineages(self):
         return
 
 
+@dataclass
+class SnowflakeSinkConnector:
+    connector_manifest: ConnectorManifest
+    report: KafkaConnectSourceReport
+
+    def __init__(
+        self,
+        connector_manifest: ConnectorManifest,
+        config: KafkaConnectSourceConfig,
+        ctx: PipelineContext,
+        report: KafkaConnectSourceReport,
+    ) -> None:
+        self.connector_manifest = connector_manifest
+        self.config = config
+        self.ctx = ctx
+        self.report = report
+        self._extract_lineages()
+        if self.config.include_column_lineage:
+            self._extract_cll()
+
+    @dataclass
+    class SnowflakeParser:
+        database_name: str
+        schema_name: str
+        topics_to_tables: Dict[str, str]
+
+    def report_warning(self, key: str, reason: str) -> None:
+        logger.warning(f"{key}: {reason}")
+        self.report.report_warning(key, reason)
+
+    def get_parser(
+        self,
+        connector_manifest: ConnectorManifest,
+    ) -> SnowflakeParser:
+        database_name = connector_manifest.config["snowflake.database.name"]
+        schema_name = connector_manifest.config["snowflake.schema.name"]
+        topics_to_tables: Dict[str, str] = {}
+
+        # If config contains topic to table map, take target table from config
+        # Else kafka connect create target table with same name as source topic
+        if "snowflake.topic2table.map" in connector_manifest.config:
+            for each in connector_manifest.config["snowflake.topic2table.map"].split(
+                ","
+            ):
+                topic, table = each.split(":")
+                # Extract lineage for only those topics whose data ingestion started
+                if topic in connector_manifest.topic_names:
+                    topics_to_tables[topic] = table
+        else:
+            for topic in connector_manifest.topic_names:
+                topics_to_tables[topic] = topic
+
+        return self.SnowflakeParser(
+            database_name=database_name,
+            schema_name=schema_name,
+            topics_to_tables=topics_to_tables,
+        )
+
+    def _extract_lineages(self):
+        self.connector_manifest.flow_property_bag = self.connector_manifest.config
+        # remove private key from properties
+        del self.connector_manifest.flow_property_bag["snowflake.private.key"]
```

Comment:
Let's have a list of secret properties and remove all of them, similar to how its done in s3 sink connector. Ideally every connector should have this property similar to connector_manifest.
`snowflake.private.key`
`snowflake.private.key.passphrase`
`value.converter.basic.auth.user.info` (from docs, it looks sensitive)